vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java

```diff
@@ -22,24 +31,64 @@
     @Autowired
     private ISpaceManager spaceManager;
 
+    @Autowired
+    private ISpaceDisplayManager spaceDisplayManager;
+
+    @Autowired
+    private AuthenticationFacade authFacade;
+
     @After("execution(public * edu.asu.diging.vspace.web..*Controller.*(..))")
     public void setExhibition(JoinPoint jp) {
         Object[] args = jp.getArgs();
         MethodSignature signature = (MethodSignature) jp.getSignature();
         Class<?> returnType = signature.getReturnType();
         // only if there is a model object injected and if we're returning a string
         // (assuming that returning a string implies rendering of a view afterwards
-        if (args != null && returnType == String.class) {
-            for (Object obj : args) {
-                if (obj instanceof Model && !(obj instanceof RedirectAttributes)) {
-                    if (!((Model) obj).containsAttribute("exhibition")) {
+        if(args!=null && returnType==String.class) {
+            for(Object obj : args) {
+                if(obj instanceof Model && !(obj instanceof RedirectAttributes)) {
+                    if(!((Model) obj).containsAttribute("exhibition")) {
                         ((Model) obj).addAttribute("exhibition", exhibitionManager.getStartExhibition());
                     }
-                    if (!((Model) obj).containsAttribute("allSpaces")) {
+                    if(!((Model) obj).containsAttribute("allSpaces")) {
                         ((Model) obj).addAttribute("allSpaces", spaceManager.getAllSpaces());
                     }
                 }
             }
         }
     }
+
+    @Around("execution(public * edu.asu.diging.vspace.web.publicview..*Controller.*(..))")
+    public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
+        Object[] args = jp.getArgs();
+        MethodSignature signature = (MethodSignature) jp.getSignature();
+        int indexofModel = (Arrays.asList(signature.getParameterTypes())).indexOf(Model.class);
+        int indexofSpaceId = (Arrays.asList(signature.getParameterTypes())).indexOf(String.class);
```

Comment:
I see two issues with this:
- this method is called for any method of controllers in the publicview package, right? so, controllers that handle the module pages will be intercepted too, but this line will return a module id not a space id
- often controller methods have several String parameters, unless the space id is the first one, this will fail. I would check the parameter name to make sure it's 'id' or 'spaceId'.